src/matrix_solver.py

```diff
@@ -12,8 +12,12 @@ def solve_matrix(A, b):
     """
 
     import numpy as np
-    from numpy.linalg import solve
+    import numpy.linalg as linalg
 
-    temperature_array = solve(A, b)
+    try:
+        i = np.linalg.inv(A)
+        temperature_array = linalg.solve(A, b)
 
-    return temperature_array
+        return temperature_array
+    except np.linalg.LinAlgError as err:
+        return False
```

Comment:
Instead of returning `False` maybe consider terminating the program and outputting an error. If done this way you will have to wrap every call to this function in something to check for this. Otherwise you may have confused developers wondering why they get a type error when trying to work with the `temperature_array` later on.

Reply:
Agreed - it is not consistent with your interface to return something of a different type than advertised.